### PR TITLE
Fix anchor issue

### PR DESCRIPTION
**Pull request by rogermparent**
This PR attempts to fix the weird "click an anchor and scroll back to top" issue that seems to only be part of cml.dev, and only when a tab is in the URL. It's also related to our slash redirection, as that being removed also remedies the issue.

https://user-images.githubusercontent.com/9111807/213613077-3a417c48-75f5-43e7-9d84-698a91ce0843.mp4

I don't 100% know _why_ this works, but it seems to do so consistently.

### CODE REVIEW

**Comment by yathomasi** [comment]
It looks like it's working fine now: https://cml.dev/doc/self-hosted-runners/?tab=GitLab

I was able to replicate the issue and I guessed [release-notes/migrating-from-v4-to-v5/#trailingslash-is-set-to-always](https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v4-to-v5/#trailingslash-is-set-to-always) could be the issue. 

Not sure if it was a cache issue or if the latest all packages upgrade fixed it.

**Comment by 0x2b3bfa0** [comment]
It works on my computer ™️

**Comment by rogermparent** [comment]
> It looks like it's working fine now: https://cml.dev/doc/self-hosted-runners/?tab=GitLab

Still get the issue on my end in production:

https://user-images.githubusercontent.com/9111807/213768256-10064e84-fd0d-4a45-83d9-e9f11f306e7c.mp4

> It works on my computer tm

@0x2b3bfa0 do you mean this PR, or production?

**Comment by 0x2b3bfa0** [comment]
> _@0x2b3bfa0 do you mean this PR, or production is working fine?_

On this pull request.

File: gatsby-config.js [hunk 1]
```diff
@@ -140,6 +140,7 @@ if (process.env.ANALYZE) {
 }
 
 module.exports = {
+  trailingSlash: 'never',
```
[the remark this answers is not in the document]
Comment (yathomasi) [reply gatsby-config.js:143]: I am not sure, do we know if we set this in the theme will it be inherited by the consumer projects?

**Review by yathomasi — APPROVED** [review]
Looks good.

File: gatsby-config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (rogermparent) [reply gatsby-config.js:143]: Good point, I thought we had it on there already but it seems like we don't. I'll try that out.

**Pull request merged**